src/Appwrite/Utopia/Request.php

```diff
@@ -95,4 +95,42 @@ public static function hasRoute(): bool
     {
         return self::$route != null;
     }
+
+    /**
+     * Get headers
+     *
+     * Method for getting all HTTP header parameters, including cookies.
+     *
+     * @return array<string,mixed>
+     */
+    public function getHeaders(): array
+    {
+        $headers = $this->generateHeaders();
+
+        $cookieHeaders = [];
+        foreach ($this->swoole->cookie as $key => $value) {
+            $cookieHeaders[] = "{$key}={$value}";
+        }
+
+        if (!empty($cookieHeaders)) {
+            $headers['cookie'] = \implode('; ', $cookieHeaders);
+        }
+
+        return $headers;
+    }
+
+    /**
+     * Get header
+     *
+     * Method for querying HTTP header parameters. If $key is not found $default value will be returned.
+     *
+     * @param  string  $key
+     * @param  string  $default
+     * @return string
+     */
+    public function getHeader(string $key, string $default = ''): string
+    {
+        $headers = $this->getHeaders();
+        return $headers[$key] ?? $default;
+    }
```

Comment:
I think we already have a getHeader method in the base class. 🤔

Reply:
We do but its using headers from swoole directly. By adding this, its using new getHeaders method, ensuring you can `getHeader("cookie")`

Reply:
Okay got it